### ISSUE TITLE
procutils.which does not find executables in path on win32/XP Prof.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7239]
|<img alt="mstarzyk's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| mstarzyk reported|
|-|-|
|Trac ID|trac#485|
|Type|defect|
|Created|2004-01-31 08:51:50Z|

<details><summary>Searchable metadata</summary>

```
trac-id__485 485
type__defect defect
reporter__mstarzyk mstarzyk
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1075539110000000 1075539110000000
changetime__1471923965328809 1471923965328809
version__ 
owner__ 
cc__exarkun cc__mstarzyk
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="mstarzyk's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|mstarzyk commented|
|-|-|

```
#!html
<pre>
Looks like os.extsep.join((some_file_wo_extension, extension)) gives
two dots instead of one. Please see below:

D:\>python
Python 2.3.3 (#51, Dec 18 2003, 20:22:39) [MSC v.1200 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.

>>> import os
>>> from twisted.runner import procutils
>>> procutils.which('twistd')
[]
>>> exts = filter(None, os.environ.get('PATHEXT', '').split(os.pathsep))
>>> exts
['.COM', '.EXE', '.BAT', '.CMD', '.VBS', '.VBE', '.JS', '.JSE', '.WSF', '.WSH']
>>> pypath = [p for p in os.environ['PATH'].split(os.pathsep) if 'py23' in p]
>>> pypath
['h:\\dev\\py23', 'h:\\dev\\py23\\scripts']
>>> twistd = os.path.join(pypath[-1], 'twistd')
>>> twistd
'h:\\dev\\py23\\scripts\\twistd'
>>> bat = exts[2]
>>> bat
'.BAT'
>>> twistd + bat
'h:\\dev\\py23\\scripts\\twistd.BAT'
>>> os.access(twistd + bat, os.X_OK)
True
>>> os.extsep.join((twistd, bat))
'h:\\dev\\py23\\scripts\\twistd..BAT'
>>> os.access(os.extsep.join((twistd, bat)), os.X_OK)
False
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
The code was written under the belief that the "." was not present in the list.
 I've changed this assumption, and the tests that went with it.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc commented|
|-|-|

In changeset defe32ce6ed4b89beb8bff10c465782019813d9b
```
#!CommitTicketReference repository="" revision="defe32ce6ed4b89beb8bff10c465782019813d9b"
Merge pull request #485 from twisted/8677-rodrigc-appveyor-iocp

Author: rodrigc
Reviewer: opalmer
Fixes: #8677

Update appveyor config to test select and iocp reactors
```
